### PR TITLE
Rename `post_only` to `secure_only` and allow for future enhancements to use bearer tokens

### PR DESCRIPTION
**Pull request by michalkleiner**
### Description:

As per the title, renaming the config and the db field to future proof the feature.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by michalkleiner** [comment]
Update relevant FAQ as well.

**Comment by michalkleiner** [comment]
@sgiehl ready for review, relevant UI test screenshots updated.

File: config/global.ini.php [hunk 1]
```diff
@@ -539,7 +539,7 @@
 ; token_auth as URL parameter in GET requests. When enabled all new authentication tokens
 ; will be created as POST only. Previously created tokens will only be accepted in POST requests.
 ; Recommended for best security.
-only_allow_posted_auth_tokens = 0
+only_allow_secure_auth_tokens = 0
```
Comment (sgiehl) [inline config/global.ini.php:542]: Should we update the description for this setting already?

File: plugins/UsersManager/lang/en.json [hunk 2]
```diff
@@ -218,9 +218,9 @@
...
-        "AuthTokenPostOnlyHelp": "Enable this option to only allow this token to be used for POST requests, this is recommended as a best security practice. The token will then not be valid as a URL parameter in GET requests.",
-        "AuthTokenPostOnlyHelpForced": "The system administrator has configured Matomo to only allow tokens to be created for use in POST requests, you cannot change this token option.",
-        "OnlyAllowPostRequests": "Only allow POST requests",
-        "PostOnly": "POST only"
+        "AuthTokenSecureOnlyHelp": "Enable this option to only allow this token to be used in a secure way (e.g. POST requests), this is recommended as a best security practice. The token will then not be valid as a URL parameter in GET requests.",
+        "AuthTokenSecureOnlyHelpForced": "The system administrator has configured Matomo to only allow tokens to be created for use in secure way (e.g. via POST requests), you cannot change this token option.",
+        "OnlyAllowSecureRequests": "Only allow secure requests",
+        "SecureOnly": "Secure only"
```
Comment (sgiehl) [inline plugins/UsersManager/lang/en.json]: Actually I'm wondering if we should improve that table headline. I doubt many people will understand what that should mean. And I also wouldn't know how to translate that when seeing it without any context.
Would `Secure use only` maybe be better?

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left some minor comments. Otherwise this looks good to merge

File: plugins/UsersManager/lang/en.json (hunk 2 above)
Comment (michalkleiner) [reply plugins/UsersManager/lang/en.json]: Yep, that sounds good, have updated.

File: config/global.ini.php (hunk 1 above)
Comment (michalkleiner) [reply config/global.ini.php:542]: Updated

**Comment by michalkleiner** [comment]
Will update the UI screenshots again to account for the changed table header and then it'll be good to merge I guess.

Have updated the FAQ as well — https://matomo.org/faq/general/faq_114/

**Review by sgiehl — APPROVED** [review]
Looks good now 👍

**Pull request merged**